baybe/insights/shap.py

```diff
@@ -0,0 +1,373 @@
+    @classmethod
+    def from_recommender(
+        cls,
+        recommender: BayesianRecommender,
+        searchspace: SearchSpace,
+        objective: Objective,
+        bg_data: pd.DataFrame,
+        explained_data: pd.DataFrame | None = None,
+        explainer_class: shap.Explainer | str = "KernelExplainer",
+        computational_representation: bool = False,
+    ):
+        """Create a SHAP insight from a recommender.
+
+        Args:
+            recommender: The recommender to be used for the SHAP insight.
+            searchspace: The searchspace for the recommender.
+            objective: The objective for the recommender.
+            bg_data: The background data set for Explainer.
+                This is also the measurement data set for the recommender.
+            explained_data: The data set to be explained. If None,
+                the background data set is used.
+            explainer_class: The explainer class.
+            computational_representation:
+                Whether to use the computational representation.
+
+        Returns:
+            SHAPInsight: The SHAP insight object.
+
+        Raises:
+            ValueError: If the recommender has not implemented a "get_surrogate" method.
+        """
+        if not hasattr(recommender, "get_surrogate"):
+            raise ValueError(
+                "The provided recommender does not provide a surrogate model."
+            )
+        surrogate_model = recommender.get_surrogate(searchspace, objective, bg_data)
+
+        return cls(
+            surrogate_model,
+            bg_data=searchspace.transform(bg_data)
+            if computational_representation
+            else bg_data,
+            explained_data=explained_data,
+            explainer_class=explainer_class,
+            computational_representation=computational_representation,
+        )
+
+    def _init_explainer(
+        self,
+        data: pd.DataFrame,
+        explainer_class: type[shap.Explainer] = shap.KernelExplainer,
+        **kwargs,
+    ) -> shap.Explainer:
+        """Create an explainer for the provided campaign.
+
+        Args:
+            data: The background data set.
+            explainer_class: The explainer class to be used.
+            **kwargs: Additional keyword arguments to be passed to the explainer.
+
+        Returns:
+            shap.Explainer: The created explainer object.
+
+        Raises:
+            NotImplementedError: If the provided explainer class does
+                not support the experimental representation.
+            ValueError: If the provided background data set is empty.
+            TypeError: If the provided explainer class does not
+                support the campaign surrogate.
+        """
+        if not self._is_shap_explainer and not self._computational_representation:
+            raise NotImplementedError(
+                "Experimental representation is not "
+                "supported for non-Kernel SHAP explainer."

```

Comment:
Question: the check above is for all non-SHAP explainers not jsut non-KernelSHAP explainers, is that intended?